datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt

```diff
@@ -41,15 +40,16 @@ class QuestionnaireFragment : Fragment() {
         val recyclerView = view.findViewById<RecyclerView>(R.id.recycler_view)
         val adapter = QuestionnaireItemAdapter(viewModel.questionnaireItemViewItemList)
         recyclerView.adapter = adapter
+    }
 
-        view.findViewById<Button>(R.id.submit).setOnClickListener {
-            val serializedResponse = FhirContext.forR4().newJsonParser()
-                .encodeResourceToString(viewModel.questionnaireResponse)
-            setFragmentResult(
-                QUESTIONNAIRE_RESPONSE_REQUEST_KEY,
-                bundleOf(QUESTIONNAIRE_RESPONSE_BUNDLE_KEY to serializedResponse)
-            )
-        }
+    // Returns the current questionnaire response to parent
+    fun returnQuestionnaireResult() {
```

Comment:
```suggestion
    fun getQuestionnaireResponse(): QuestionnaireResponse {
```

just return the response directly. we can stop using the results API now if the submit action is initiated outside of the fragment.

Reply:
done